src/gui/screen_printing.cpp

```diff
@@ -261,12 +266,15 @@ void screen_printing_data_t::enable_tune_button() {
     btn_tune.ico.Invalidate();
 }
 
-void screen_printing_data_t::update_remaining_time(time_t rawtime, uint16_t print_speed) {
-    w_etime_value.color_text = rawtime != time_t(-1) ? GuiDefaults::COLOR_VALUE_VALID : GuiDefaults::COLOR_VALUE_INVALID;
-    if (rawtime != time_t(-1)) {
-        if (print_speed != 100)
+void screen_printing_data_t::update_remaining_time(uint32_t millis, uint16_t print_speed) {
+    bool is_time_valid = int(millis) > 0;
```

Comment:
`bool is_time_valid = millis > 0U;` ... if there was a sign comparison warning (which I doubt)?

Reply:
but probably the whole unsigned int is a problem - either compare millis > 2^31 or keep millis as signed int32_t

Reply:
changed, must be shorter than 365 days